doc/rest-api.md

```diff
@@ -2424,3 +2427,236 @@ Input (none at present):
             }
         }
     }
+## `/1.0/storage-pools`
+### GET
+ * Description: list of storage pools
+ * Introduced: with API extension `storage`
+ * Authentication: trusted
+ * Operation: sync
+ * Return: list of storage pools that are currently defined on the host
+
+    [
+        "/1.0/storage-pools/default",
+        "/1.0/storage-pools/pool1"
+        "/1.0/storage-pools/pool2"
+        "/1.0/storage-pools/pool3"
+        "/1.0/storage-pools/pool4"
+    ]
+
+### POST
+ * Description: create a new storage pool
+ * Introduced: with API extension `storage`
+ * Authentication: trusted
+ * Operation: sync
+ * Return: standard return value or standard error
+
+Input:
+
+    {
+        "config": {
+            "size": "10GB"
+        },
+        "driver": "zfs",
+        "name": "pool1"
+    }
+
+## `/1.0/cluster`
+### GET (optional `?password=<trust-password>`)
```

Comment:
password will most likely be dropped (see other comments)

Reply:
Do you mean we should just drop it from the doc? (although the current implementation does that)

Reply:
We should drop that password field from the API too. Our workflow should not involve passing a password as an API argument.

Reply:
The password query parameter was dropped, as per the new API layout/semantics.